sbt-scalajs-bundler/src/main/scala/scalajsbundler/sbtplugin/ScalaJSBundlerPlugin.scala

```diff
@@ -64,6 +65,27 @@ object ScalaJSBundlerPlugin extends AutoPlugin {
   object autoImport {
 
     /**
+      * The BundlingMode choice dictates how webpack task is set up.
+      */
+    sealed trait BundlingMode
+
+    object BundlingMode {
+      /** Bundle the whole application */
+      case object Application extends BundlingMode
+      /** Bundle the dependencies of the application */
+      sealed trait Library extends BundlingMode {
+        /** Name of the global variable containing the dependencies */
+        def exportedName: String
+      }
+      /** Bundle only the libraries used by the application. */
+      case class LibraryOnly(exportedName: String = defaultLibrariesExportedName) extends Library
+      /** Bundle the libraries used by the application and concatenates the output of Scala.js to that bundle */
+      case class LibraryAndApplication(exportedName: String = defaultLibrariesExportedName) extends Library
+
+      val defaultLibrariesExportedName = "ScalaJSBundlerLibraries"
+    }
```

Comment:
Would it be possible to define this datatype outside of the `sbtplugin` package and then alias it (both the trait and the companion object) in `autoImport`?

Reply:
Probably? Why though?

Reply:
This file is already quite long, but the main motivation is to move as many things as possible out of sbt, so that if one day we want to create a different API (e.g. a cbt plugin, or just a CLI) that will be easier.